### PR TITLE
Feature/cplp 2142/add status subscription

### PR DESCRIPTION
**Pull request by nidhigarg-bmw**

### CODE REVIEW

File: cx-portal-shared-components/src/components/basic/OrderStatusButton/index.tsx [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+        sx={{
+          width: '100%',
+          position: 'relative',
+          zIndex: '5',
+        }}
+        {...props}
+      >
+        {label}
+      </Button>
+      {fetchButton(4, buttonStyle.background1, status === 'INACTIVE', {
```
Comment (manojava-gk) [inline cx-portal-shared-components/src/components/basic/OrderStatusButton/index.tsx:172]: Why not a loop?

File: cx-portal-shared-components/src/components/basic/OrderStatusButton/index.tsx [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+          color: color,
+          background1: '#e1e1e1',
+          background2: '#f3f3f3',
+          background3: '#f9f9f9',
+        })
+        break
+      case 'ACTIVE':
+        setButtonStyle({
+          color: 'success',
+          background1: '#f5f9ee',
```
Comment (manojava-gk) [inline cx-portal-shared-components/src/components/basic/OrderStatusButton/index.tsx:71]: same color codes are being used in multiple places. Better to move them to theme file or create a enum or const in the component itself

File: cx-portal/src/features/apps/apiSlice.ts [hunk 3]
```diff
@@ -141,7 +141,23 @@ export const apiSlice = createApi({
...
-      query: () => `/api/apps/active`,
+      async queryFn(_arg, _queryApi, _extraOptions, fetchWithBQ) {
+        const activeApps = await fetchWithBQ(`/api/apps/active`)
+        if (activeApps.error) return { error: activeApps.error }
+        const data: any = activeApps.data
+        const subscriptionStatus = await fetchWithBQ(
+          `/api/apps/subscribed/subscription-status`
+        )
+        const subscriptionData: any = subscriptionStatus.data
+        data.forEach(async (appItem: any) => {
```
Comment (manojava-gk) [inline cx-portal/src/features/apps/apiSlice.ts:152]: Can avoid any?

File: cx-portal/src/features/apps/apiSlice.ts [hunk 4]
```diff
@@ -141,7 +141,23 @@ export const apiSlice = createApi({
       query: (id: string) => `/api/apps/${id}?lang=${i18next.language}`,
     }),
     fetchActiveApps: builder.query<AppMarketplaceApp[], void>({
-      query: () => `/api/apps/active`,
+      async queryFn(_arg, _queryApi, _extraOptions, fetchWithBQ) {
+        const activeApps = await fetchWithBQ(`/api/apps/active`)
+        if (activeApps.error) return { error: activeApps.error }
+        const data: any = activeApps.data
```
Comment (manojava-gk) [inline cx-portal/src/features/apps/apiSlice.ts:147]: Can avoid any?

**Review by manojava-gk — COMMENTED** [review]
Type and Component behaviour

File: cx-portal/src/features/apps/apiSlice.ts (hunk 4 above)
Comment (oyo) [inline cx-portal/src/features/apps/apiSlice.ts:147]: we know the data type from the api endpoint here: https://portal-swagger.dev.demo.catena-x.net/?urls.primaryName=Portal%20App%20Marketplace%20API#/Apps/get_api_Apps_active
don't we have a type for this defined? please define and use

File: cx-portal/src/features/apps/apiSlice.ts [hunk 5]
```diff
@@ -141,7 +141,23 @@ export const apiSlice = createApi({
...
+        const subscriptionStatus = await fetchWithBQ(
+          `/api/apps/subscribed/subscription-status`
+        )
+        const subscriptionData: any = subscriptionStatus.data
```
Comment (oyo) [inline cx-portal/src/features/apps/apiSlice.ts:151]: please define and use type

**Review by oyo — CHANGES_REQUESTED** [review]

**Review by oyo — DISMISSED** [review]
looks good to me now

**Review by manojava-gk — APPROVED** [review]
looks okay now

**Pull request merged**